api/src/services/relations.ts

```diff
@@ -112,7 +112,7 @@ export class RelationsService {
 			return metaRow.many_collection === collection;
 		});
 
-		const schemaRows = await this.foreignKeys(collection);
+		const schemaRows = bypassCache ? await this.schemaInspector.foreignKeys() : await this.foreignKeys(collection);
```

Comment:
```suggestion
		const schemaRows = bypassCache ? await this.schemaInspector.foreignKeys().filter((row) => row.table === collection) : await this.foreignKeys(collection);
```

Reply:
This would also be inaccurate, as there should be no filtering if `collection` is not defined. Here's [the PR](https://github.com/directus/directus/pull/19391) which changed this behaviour.